ingest/src/withConfig.ts

```diff
@@ -0,0 +1,88 @@
+import yargs from "yargs";
+import Path from "path";
+import { Config, Constructor } from "./Config";
+
+export type LoadConfigArgs = {
+  config?: string;
+};
+
+export const loadConfig = async ({
+  config: configPathIn,
+}: LoadConfigArgs): Promise<Config> => {
+  const path = Path.resolve(
+    configPathIn === undefined ? "ingest.config.cjs" : configPathIn
+  );
+
+  const configArray = (await import(path)).default as Config[];
+
+  // Validate config
+  if (!Array.isArray(configArray)) {
+    throw new Error(
+      `Invalid config at ${path}: expected default exported array of Config objects`
+    );
+  }
+  // TODO: further validation
+
+  // Flatten config
+  const config: Config = configArray.reduce((acc, cur) => ({
+    ...acc,
+    ...cur,
+  }));
+
+  return config;
+};
+
+export const withConfig = async <T>(
+  action: (config: ResolvedConfig, args: T) => Promise<void>,
+  args: LoadConfigArgs & T
+) => {
+  const config = await loadConfig(args);
+  const resolvedConfig = await resolveConfig(config);
+
+  console.log(`resolved config: ${resolvedConfig.dataSources[0]}`);
+  try {
+    return await action(resolvedConfig, args);
+  } finally {
+    // TODO
+    // config.embeddedContentStore.close();
```

Comment:
all the stores need to be closed, right?

Reply:
Correct